db/products-controllers.ts

```diff
@@ -0,0 +1,94 @@
+import Product from './products';
+import * as productValidation from '../config/products-validation';
+import {logger} from '../config/logger';
+
+class ProductsControllers {
+    async getProducts(ctx) {
+        ctx.ok(await Product.find().select('-__v'));
+    }
+
+    async getProductById(ctx) {
+        try {
```

Comment:
read about middlewares and create looger middlewarre and catch errors middleware (should catch your exceptions)